### PR TITLE
The toURI method returns null if the path is not absolute.

### PR DESCRIPTION
**Pull request by pkriens**

### CODE REVIEW

**Comment by iloveeclipse** [comment]
@pkriens : could you please add a line to the commit message with this content:

`Fixes https://github.com/eclipse-platform/eclipse.platform/issues/1114`

This would automatically link this PR with the bug, fix the bug on merge and allow easy navigation from IDE to github?

File: resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileStore.java [hunk 1]
```diff
@@ -32,11 +32,12 @@ public class NullFileStore extends FileStore {
 
 	/**
 	 * Creates a null file store
+	 * 
 	 * @param path The path of the file in this store
 	 */
 	public NullFileStore(IPath path) {
 		Assert.isNotNull(path);
-		this.path = path;
+		this.path = path.makeAbsolute();
```
Comment (laeubi) [inline resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileStore.java:40]:
```suggestion
		Assert.isLegal(path.isAbsolute());
```

Comment (pkriens) [reply resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileStore.java:40]: It is a null file system so you only reduce its input domain for no reason whatsoever.

Comment (laeubi) [reply resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileStore.java:40]: Passing a non absolute path seems illegal for this (internal) constructor, so it should be asserted but not fix caller errors automatically.

**Comment by pkriens** [comment]
@iloveeclipse I did and get weird errors in the build now. Also had rather puzzling review from @laeubi 

You can take this as you please or close it but this is more time than I wanted to spend on this.

**Comment by iloveeclipse** [comment]
The small fix should be fine, if it allows https://github.com/eclipse-platform/eclipse.platform/issues/1114 to be fixed. There aren't so many users of this, and @pkriens contribution makes sense for them.

**Comment by pkriens** [comment]
@iloveeclipse thanks.

**Pull request merged**